src/understand-tidb/transaction.md

```diff
@@ -1 +1,114 @@
 # Transaction
+
+The transaction engine in TiDB is responsible to provide `ACID` guarantee for all the read and write requests. It consists of the client/coordinator part in `tidb` repository and the server/participant part in `tikv` repository. This document is mainly about the `tidb` part.
+
+# The Architecture
+
+In `TiDB` the transaction write flow is like this:
+
+![transaction-architecture](../img/transaction-architecture.png)
+
+After the transaction is started in a session, all the reads and writes will use a snapshot to fetch data, and the written content will be buffered in the memory
+of the transaction. When the `commit` statement is received from the client, the `percolator` protocol will be used to commit these changes to the storage system.
+
+# The Interface
+
+In `tidb`, the [Transaction](https://github.com/pingcap/tidb/blob/af70762cd52519f025daa5e869ba37465a7fb311/kv/kv.go#L181) interface defines the commonly used transaction operations.
+
+```go
+type Transaction interface {
+  ...
+	// Commit commits the transaction operations to KV store.
+	Commit(context.Context) error
+	// Rollback undoes the transaction operations to KV store.
+	Rollback() error
+	// LockKeys tries to lock the entries with the keys in KV store.
+	// Will block until all keys are locked successfully or an error occurs.
+	LockKeys(ctx context.Context, lockCtx *LockCtx, keys ...Key) error
+	// SetOption sets an option with a value, when val is nil, uses the default
+	// value of this option.
+	SetOption(opt int, val interface{})
+	// GetOption returns the option
+	GetOption(opt int) interface{}
+	...
+	// StartTS returns the transaction start timestamp.
+	StartTS() uint64
+	// Valid returns if the transaction is valid.
+	// A transaction become invalid after commit or rollback.
+	Valid() bool
+	// GetMemBuffer return the MemBuffer binding to this transaction.
+	GetMemBuffer() MemBuffer
+	// GetSnapshot returns the Snapshot binding to this transaction.
+	GetSnapshot() Snapshot
+  ...
+}
+```
+
+These are common interfaces the transaction will provide, for example `Commit` will be used to commit the current ongoing transaction. The transaction is considered ongoing before the `commit` operations is triggered. The two phase commit processing will be used to commit a transaction and it will finally become committed or aborted. 
```

Comment:
```suggestion
These are common interfaces the transaction will provide. For example, `Commit` will be used to commit the current ongoing transaction. The transaction is considered ongoing before the `Commit` operation is triggered. The two-phase commit processing will be used to commit a transaction and it will finally become committed or aborted. 
```